baremetal/metal3machine_manager.go

```diff
@@ -843,26 +812,22 @@ func (m *MachineManager) chooseHost(ctx context.Context) (*bmov1alpha1.BareMetal
 			}
 		}
 
-		if labelSelector.Matches(labels.Set(host.ObjectMeta.Labels)) {
```

Comment:
Why are you removing` labelSelector.Matches(labels.Set(host.ObjectMeta.Labels))` condition? Could you elaborate, please?